preview-src/comment.tsx

```diff
@@ -0,0 +1,159 @@
+import * as React from 'react';
+import { useContext, useState, useEffect, useRef } from 'react';
+
+import Markdown from './markdown';
+import { Spaced, nbsp } from './space';
+import { Avatar, AuthorLink } from './user';
+import Timestamp from './timestamp';
+import { Comment } from '../src/common/comment';
+import { PullRequest } from './cache';
+import PullRequestContext from './context';
+import { editIcon, deleteIcon } from './icon';
+
+export type Props = Partial<Comment & PullRequest>;
+
+export function CommentView(comment: Props) {
+	const { id, pullRequestReviewId, canEdit, canDelete, user, author, htmlUrl, createdAt, bodyHTML, body, } = comment;
+	const [ bodyMd, setBodyMd ] = useState(body);
+	const { deleteComment, editComment, pr } = useContext(PullRequestContext);
+	const currentDraft = pr.pendingCommentDrafts && pr.pendingCommentDrafts[id];
+	const [inEditMode, setEditMode] = useState(!!currentDraft);
+	const [showActionBar, setShowActionBar] = useState(false);
+
+	useEffect(() => {
+		if (body !== bodyMd) {
+			setBodyMd(body);
+		}
+	}, [body]);
+
+	if (inEditMode) {
+		return <EditComment id={id}
+			body={currentDraft || body}
+			onCancel={
```

Comment:
If I start editing something, make some changes, and then click cancel, if I click edit again the text shown is my edited text and not the current text. Do we need to clear the pending comment cache at this point?

Reply:
One other issue with pending comments - if you start editing the description on one PR, and then without closing the edit component, switch to another, the edit component is still rendered on that one with the old description

Reply:
Fixed.